tests/unit/plugins/listops/dictionary_operation.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2018 Weile Wei
+// Copyright (c) 2018 Parsa Amini
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/phylanx.hpp>
+#include <phylanx/plugins/listops/dictionary_operation.hpp>
+
+#include <hpx/hpx_main.hpp>
+#include <hpx/include/lcos.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <cstdint>
+#include <string>
+#include <utility>
+#include <vector>
+
+//////////////////////////////////////////////////////////////////////////
+phylanx::execution_tree::primitive_argument_type compile_and_run(
+    std::string const& codestr)
+{
+    phylanx::execution_tree::compiler::function_list snippets;
+    phylanx::execution_tree::compiler::environment env =
+        phylanx::execution_tree::compiler::default_environment();
+
+    auto const& code = phylanx::execution_tree::compile(codestr, snippets, env);
+    return code.run();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+void test_dictionary_operation(std::string const& code,
+    std::string const& expected_str)
+{
+    HPX_TEST_EQ(compile_and_run(code), compile_and_run(expected_str));
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+void test_dict_operation()
+{
+    char const* const code = "'('(1, 2), '(3, 4))";
```

Comment:
Please use `list()` instead of `'()`. We will remove the quoting operator in the future.